synapse/handlers/federation.py

```diff
@@ -946,6 +946,12 @@ async def on_invite_request(
         )
 
         context = EventContext.for_outlier(self._storage_controllers)
+
+        if event.is_notifiable:
+            await self._federation_event_handler.bulk_push_rule_evaluator.action_for_event_by_user(
```

Comment:
We'd prefer if you could add a `self._bulk_push_rule_evaluator = hs.get_bulk_push_rule_evaluator()` to this class than reaching into another's attributes.